src/data_source/update.rs

```diff
@@ -126,3 +111,21 @@ where
         Ok(())
     }
 }
+
+/// A data source with an atomic transaction-based synchronization interface.
+#[async_trait]
+pub trait VersionedDataSource {
+    type Error: Error + Debug;
+
+    /// Atomically commit to all outstanding modifications to the data.
+    ///
+    /// If this method fails, outstanding changes are left unmodified. The caller may opt to retry
+    /// or to erase outstanding changes with [`revert_version`](Self::revert_version).
+    async fn commit_version(&mut self) -> Result<(), Self::Error>;
```

Comment:
Since for postgres we can't revert back to older versions once we commit, should would it make more sense to call the trait functions `commit` and `revert` instead?

Reply:
Perhaps. `atomic_store` doesn't let you revert to old versions either, so the behavior isn't really any different. But I think your names are better/less confusing